### PR TITLE
Switched to 2 dirs for single/multi node VMs and updated readme

### PR DESCRIPTION
**Pull request by ankanm**
Issue #4 - Split up single/multi node vagrant dirs.
Issue #2 - Updated documentation and updating cache at start of base.

@miroswan @jordmoz @vmahedia

### CODE REVIEW

File: ansible/roles/base/tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,11 @@
 ---
```
Comment (miroswan) [inline ansible/roles/base/tasks/main.yml]: Good call. I ran into this myself.

File: vagrant/multi_node/hosts/localmesos-cluster [hunk 2]
```diff
@@ -1,25 +1,25 @@
 #inventory file for local multi-node cluster environment
 
 [machines]
```
Comment (miroswan) [inline vagrant/multi_node/hosts/localmesos-cluster]: Good call. This way /etc/hosts doesn't have to change when going between single and mutli

File: ansible/roles/base/tasks/main.yml [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+- name: Update package repo cache
+  apt: update_cache=yes
+  sudo: yes
```
Comment (miroswan) [inline ansible/roles/base/tasks/main.yml]: Good call, I ran into this myself

File: vagrant/multi_node/hosts/localmesos-cluster [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-localmesos[01:03] env=test base_data_dir=/data shared_mount=/vagrant
+localmesos[02:04] env=test base_data_dir=/data shared_mount=/vagrant
```
Comment (miroswan) [inline vagrant/multi_node/hosts/localmesos-cluster]: Good call. This way /etc/hosts doesn't have to change when going between single and mutli

**Comment by miroswan** [comment]
I'm going to go through the provisioning as a user for both single and multi and see if everything lines up.

**Comment by miroswan** [comment]
Seeing this issue on single node setup:

```
Demitris-MacBook-Pro-3:single_node demitri$ ./setup-single-node.sh
Kernel reload()...
SSH password:

PLAY [machines] ***************************************************************

GATHERING FACTS ***************************************************************
fatal: [localmesos01] => {'msg': 'FAILED: [Errno 8] nodename nor servname provided, or not known', 'failed': True}

TASK: [base | Update package repo cache] **************************************
FATAL: no hosts matched or all hosts have already failed -- aborting


PLAY RECAP ********************************************************************
           to retry, use: --limit @/Users/demitri/base.retry

localmesos01               : ok=0    changed=0    unreachable=1    failed=0
```

I passed it the standard vagrant password (vagrant) without much luck. Was there a different password in use?

**Comment by miroswan** [comment]
Noted that the main ReadMe.md should probably be updated to reference the two dirs.

**Comment by miroswan** [comment]
After adding the host to /etc/hosts on my mac (it _was_ in the instructions), the provisioning script started to work, but I think your hostsupdater plugin should be handling this for you.

**Comment by miroswan** [comment]
Single node provisioned correctly.

**Comment by miroswan** [comment]
Didn't have to setup /etc/hosts in mutli mode, which is pretty good.

**Comment by miroswan** [comment]
Multi-node worked as expected. Hit 8080, 5050, and 8000 on the three nodes. Redirection to leader worked when hitting Mesos.

**Comment by ankanm** [comment]
It's odd for that to happen with the single node. I believe it's intermittent. I destroyed then confirmed that /etc/hosts doesn't have an entry (so as to start from scratch) and it worked.
I'm going to merge this given that @miroswan has reviewed and tested it @vmahedia.

**Pull request merged**